File: LeetCode/961.rs
fn repeated_n_times(nums: Vec<i32>) -> i32 {
    let mut nums = nums;
    let n = nums.len();
    nums.sort_unstable();
    if nums[n / 2] == nums[n - 1] {
        nums[n / 2]
    } else {
        nums[n / 2 - 1]
    }
}

fn main() {
    assert_eq!(3, repeated_n_times(vec![1, 2, 3, 3]));
    assert_eq!(3, repeated_n_times(vec![5, 4, 3, 3]));
    assert_eq!(2, repeated_n_times(vec![2, 1, 2, 5, 3, 2]));
    assert_eq!(5, repeated_n_times(vec![5, 1, 5, 2, 5, 3, 5, 4]));
}

File: LeetCode/961.cpp
#include <iostream>
#include <random>
#include <vector>
using namespace std;

int repeatedNTimes(vector<int> nums) {
    int n = nums.size();
    mt19937 gen{random_device{}()};
    uniform_int_distribution<int> dis(0, n - 1);

    while (true) {
        int x = dis(gen), y = dis(gen);
        if (x != y && nums[x] == nums[y]) {
            return nums[x];
        }
    }
}

#define ASSERT(except, actual)                                                 \
    {                                                                          \
        auto res = actual;                                                     \
        if (except != res) {                                                   \
            cout << #actual << "\nexcept: " #except ", but found: " << res     \
                 << "\n";                                                      \
        }                                                                      \
    }

int main() {
    ASSERT(3, repeatedNTimes({1, 2, 3, 3}));
    ASSERT(3, repeatedNTimes({5, 4, 3, 3}));
    ASSERT(2, repeatedNTimes({2, 1, 2, 5, 3, 2}));
    ASSERT(5, repeatedNTimes({5, 1, 5, 2, 5, 3, 5, 4}));
}